core/src/main/java/org/renjin/sexp/Environment.java

```diff
@@ -277,32 +291,36 @@ public void setVariable(Context context, Symbol symbol, SEXP value) {
       throw new EvalException("cannot add bindings to a locked environment");
     }
     if(activeBindings != null && activeBindings.containsKey(symbol)) {
-      callActiveBinding(context, symbol, value);
+      Closure fun = activeBindings.get(symbol);
+      Environment environment = fun.getEnclosingEnvironment();

```

Comment:
Is this the correct environment for evaluation? Add a test case with a active binding function that calls `parent.frame()` to find out.